src/proxy.rs

```diff
@@ -188,7 +198,8 @@ impl Proxy<ConnectedState> {
         let mut proxy_to_server = {
             // TODO(#144): Read options from config
             let options = client::Options::default();
-            Client::new(options)
+            let client = Client::new(options);
+            ImapState::client(client)
```

Comment:
I'm not sure `ImapState` is the best name. Something with `Wrapper` would probably make the intention more clear.

Reply:
I don't like the name either. I'm open to suggestions.

Reply:
How about... `FragmentTracker`, `LogWrapper`, `FragmentLogger`, `...`?

Reply:
It's called `FragmentTracker` now.